### PR TITLE
Remove Kubefed from Operator

### PR DESCRIPTION
**Pull request by aswinsuryan**
Remove Kubefed from Operator

### CODE REVIEW

File: pkg/broker/rbac.go [hunk 1]
```diff
@@ -66,6 +66,11 @@ func NewBrokerAdminRole() *rbacv1.Role {
 				APIGroups: []string{"rbac.authorization.k8s.io"},
 				Resources: []string{"rolebindings"},
 			},
+			rbacv1.PolicyRule{
```
Comment (mangelajo) [inline pkg/broker/rbac.go:69]: This role is for "subctl" only, if you want the clusters to be able to do this, add it to the NewBrokerClusterRole only

File: pkg/broker/rbac.go [hunk 2]
```diff
@@ -66,6 +66,11 @@ func NewBrokerAdminRole() *rbacv1.Role {
...
+				Verbs:     []string{"create", "get", "list", "watch", "patch", "update", "delete"},
+				APIGroups: []string{"lighthouse.submariner.io"},
+				Resources: []string{"multiclusterservices"},
+			},
```
Comment (mangelajo) [inline pkg/broker/rbac.go:73]:
```suggestion
```

File: pkg/controller/servicediscovery/servicediscovery_controller.go [hunk 3]
```diff
@@ -176,7 +176,7 @@ func newLighthouseAgent(cr *submarinerv1alpha1.ServiceDiscovery) *appsv1.Deploym
 						},
 					},
 
-					ServiceAccountName:            "submariner-operator",
+					ServiceAccountName:            "submariner-lighthouse",
```
Comment (mangelajo) [inline pkg/controller/servicediscovery/servicediscovery_controller.go:179]: Nice, we should do more of this for the other services, scoping to their specific permissions/service account :+1:

File: pkg/engine/crds.go [hunk 4]
```diff
@@ -41,6 +41,10 @@ func Ensure(config *rest.Config) error {
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return fmt.Errorf("error creating the Endpoint CRD: %s", err)
 	}
+	_, err = apiext.ApiextensionsV1beta1().CustomResourceDefinitions().Create(getMcsCRD())
```
Comment (mangelajo) [inline pkg/engine/crds.go]: please use utils.CreateOrUpdateCrd, see
https://github.com/submariner-io/submariner-operator/commit/753b7003b4b02905815fd68da660dc0e1908067c#diff-07a74df9a382711c002273a217a37de4R50

File: pkg/broker/rbac.go (hunk 1 above)
Comment (aswinsuryan) [reply pkg/broker/rbac.go:69]: Without this admin was not able to grant cluster roles and deployment was failing hence added this one.

File: pkg/engine/crds.go (hunk 4 above)
Comment (aswinsuryan) [reply pkg/engine/crds.go]: Done , this has been now moved to pkg/lighthouse/crds.go, made the changes there.

File: pkg/broker/rbac.go (hunk 1 above)
Comment (mangelajo) [reply pkg/broker/rbac.go:69]: But that's unrelated, I will investigate this afterwards. I added this: https://github.com/submariner-io/submariner-operator/issues/326

**Comment by vthapar** [comment]
LGTM

**Pull request merged**